atomics/CMakeLists.txt

```diff
@@ -23,9 +23,44 @@ endif()
 blt_add_library(NAME desul_atomics
                 HEADERS include/desul/atomics.hpp
                 ${SOURCES_ARG}
-                INCLUDES include
                 ${DEPENDS_ARG})
 
-# TODO: Add tests and install targets if needed.
+
+if(ENABLE_CUDA)
+  target_include_directories(desul_atomics
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
+endif()
+
+if (ENABLE_HIP)
+  target_include_directories(desul_atomics
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
+endif()
```

Comment:
Why do we need these? It looks like the one below adds the same directory as these two do.

Reply:
INTERFACE directories are only used for targets that depend on the target, but not the target itself. Without the PUBLIC include, the source files would fail to find the includes (it has to be written this way since you cannot set PUBLIC include directories on an INTERFACE target) - and desul_atomics is an interface target when built without cuda/hip.

Reply:
I see now, I had missed the PUBLIC vs INTERFACE distinction.  That means it's appropriate to split these out, but should we have it as both public and interface or just one or the other?